bokehjs/src/coffee/client.coffee

```diff
@@ -384,32 +384,37 @@ class ClientSession
     @request_server_info().then((ignored) -> undefined)
 
   _should_suppress_on_change : (patch, event) ->
-    if event instanceof ModelChangedEvent
+    if event['kind'] == 'ModelChanged'
       for event_json in patch.content['events']
-        if event_json['kind'] == 'ModelChanged' and event_json['model']['id'] == event.model.id and event_json['attr'] == event.attr
+        if event_json['kind'] == 'ModelChanged' and event_json['model']['id'] == event['model']['id'] and event_json['attr'] == event['attr']
           patch_new = event_json['new']
-          if event.new_ instanceof HasProps
-            if typeof patch_new == 'object' and 'id' of patch_new and patch_new['id'] == event.new_.id
+          if typeof event['new'] == 'object'
+            if typeof patch_new == 'object' and 'id' of patch_new and patch_new['id'] == event['new']['id']
               return true
-          else if _.isEqual(patch_new, event.new_)
+          else if _.isEqual(patch_new, event['new'])
             return true
-    else if event instanceof RootAddedEvent
+    else if event['kind'] == 'RootAdded'
         for event_json in patch.content['events']
-          if event_json['kind'] == 'RootAdded' and event_json['model']['id'] == event.model.id
+          if event_json['kind'] == 'RootAdded' and event_json['model']['id'] == event['model']['id']
             return true
-    else if event instanceof RootRemovedEvent
+    else if event['kind'] == 'RootRemoved'
         for event_json in patch.content['events']
-          if event_json['kind'] == 'RootRemoved' and event_json['model']['id'] == event.model.id
+          if event_json['kind'] == 'RootRemoved' and event_json['model']['id'] == event['model']['id']
             return true
-    else if event instanceof TitleChangedEvent
+    else if event['kind'] == 'TitleChanged'
         for event_json in patch.content['events']
-          if event_json['kind'] == 'TitleChanged' and event_json['title'] == event.title
+          if event_json['kind'] == 'TitleChanged' and event_json['title'] == event['title']
             return true
 
     return false
 
   _document_changed : (event) ->
-    if @_current_patch? and @_should_suppress_on_change(@_current_patch, event)
+    # Filter out events that were initiated by the ClientSession itself
+    if event.setter_id == @id
+      return
+
+    json_patch = @document.create_json_patch([event])
+    if @_current_patch? and @_should_suppress_on_change(@_current_patch, json_patch['events'][0])
```

Comment:
is it actually possible to need this _should_supress_on_change code now? I guess it can only happen if 1) we have a current patch, which means we have a set() called by us on the stack with our setter_id, and 2) someone called set separately from our own set, and 3) they set the thing to same value that we did ? Pretty far-fetched right? Maybe we can delete all of this and rely on setter_id instead?